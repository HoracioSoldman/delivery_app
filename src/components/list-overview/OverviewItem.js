import React from 'react'
import {OverviewItemStyle} from '../stylus'
import { connect } from 'react-redux';
import {filterTable} from '../../redux/actions/shipment-actions'
import PropTypes from 'prop-types'

 function OverviewItem(props) {
    const {title, nbr, primaryColor, secondaryColor} = props.item;

    const onOverviewClicked = () =>{
        props.filterTable(title);
    }
    return (
        <OverviewItemStyle onClick={onOverviewClicked}  
            primaryColor={primaryColor} secondaryColor={secondaryColor}>
            <h4>{title}</h4>
            <p>{nbr}</p>
        </OverviewItemStyle>
    )
}

OverviewItem.propTypes = {
    item: PropTypes.shape({
        title: PropTypes.string.isRequired,
        nbr: PropTypes.number.isRequired,
        primaryColor: PropTypes.string.isRequired,
        secondaryColor: PropTypes.string.isRequired,
    })
}

export default connect(null, {filterTable})(OverviewItem);

export {OverviewItem};