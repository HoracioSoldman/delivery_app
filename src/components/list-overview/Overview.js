import React from 'react'
import {OverviewWrapper} from '../stylus'
import OverviewItem from './OverviewItem'

export default function Overview(props) {

    return (
        <OverviewWrapper>
            {
                 props.items.map((ovv, i)=>(
                    <OverviewItem item={ovv} key={i}/>
                ))
            }

        </OverviewWrapper>
    )
}
