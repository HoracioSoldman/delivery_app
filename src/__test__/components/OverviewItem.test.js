import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {OverviewItem} from '../../components/list-overview/OverviewItem'

describe('Overviewitem', () => {

    it('renders without crashing', ()=> {
        const props = {
            item:{
                title: 'ASSIGNED',
                nbr: 18,
                primaryColor: '#2196F3',
                secondaryColor: '#42A5F5'
            }
        }
        const wrapper = shallow(<OverviewItem  {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot() 
    })

    
    it('renders without crushing', ()=>{
        const props = {
            item:{
                title: 'ASSIGNED',
                nbr: 18,
                primaryColor: '#2196F3',
                secondaryColor: '#42A5F5'
            }
        }

        const wrapper = shallow(<OverviewItem { ...props} />)
        expect(wrapper.find('h4').text()).toEqual('ASSIGNED'); 
        expect(wrapper.find('p').text()).toEqual('18'); 
    }) 
})