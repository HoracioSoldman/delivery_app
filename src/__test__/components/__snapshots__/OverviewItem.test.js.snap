// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overviewitem renders without crashing 1`] = `
<styled.div
  onClick={[Function]}
  primaryColor="#2196F3"
  secondaryColor="#42A5F5"
>
  <h4>
    ASSIGNED
  </h4>
  <p>
    18
  </p>
</styled.div>
`;
